data-prepper-plugins/avro-codecs/src/test/java/org/opensearch/dataprepper/plugins/codec/avro/AvroInputCodecTest.java

```diff
@@ -166,18 +168,40 @@ private static Object decodeOutputIfEncoded(Map encodedOutput, Object key){
         }
     }
 
-    private static Event getEvent(int index){
+    private static Event getEvent(int index) throws Exception {
         List<GenericRecord> recordList=generateRecords(parseSchema(),numberOfRecords);
         GenericRecord record=recordList.get(index);
-        Schema schema=parseSchema();
+        Schema schema = record.getSchema();
+        final Map<String, Object> eventData = convertRecordToMap(record, schema);
+        final Event event = JacksonLog.builder().withData(eventData).build();
+        return event;
+    }
+
+    private static Map<String, Object> convertRecordToMap(GenericRecord record, Schema schema) throws Exception {
```

Comment:
You are duplicating the logic from `AvroInputCodec` in the test. This is not a valid test scenario. You should start with known data coming in and verify it against expected data going out.

Please remove this entire function - `convertRecordToMap`.

Reply:
Okay...

Reply:
It is done @dlvenable !